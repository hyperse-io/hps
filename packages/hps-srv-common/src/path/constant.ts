export const baseURL = 'http://urlschema.org/';
