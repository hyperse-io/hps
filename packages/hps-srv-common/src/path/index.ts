export * from './ensure-slash.js';
export * from './get-pacakge-dir.js';
export * from './get-query-string.js';
export * from './get-url-root.js';
export * from './get-url-search-param-str.js';
export * from './is-http-url.js';
export * from './modify-query-string.js';
export * from './normalize-platform-path.js';
export * from './normalize-url.js';
export * from './remove-query-string.js';
export * from './url-join.js';
