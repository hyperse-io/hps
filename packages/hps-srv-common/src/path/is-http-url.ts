/**
 * check if given url is http url or path
 * @param url url
 */
export const isHttpUrl = (url: string): boolean => {
  return url
    ? url.startsWith('//') || /(?:https|http|ftp)?:\/\//.test(url)
    : false;
};
