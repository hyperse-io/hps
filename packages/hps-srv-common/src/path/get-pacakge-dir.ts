import { join } from 'node:path';
import { getDirname } from '@armit/file-utility';
import { searchPackageDir } from '@armit/package';

export const getPackageDir = (...paths: string[]) => {
  const dir = getDirname(import.meta.url);
  const packageDir = searchPackageDir({
    cwd: dir,
  });
  if (!packageDir) {
    throw new Error('Could not resolve package root for `flatjs/evolve`');
  }
  return join(packageDir, ...paths);
};
