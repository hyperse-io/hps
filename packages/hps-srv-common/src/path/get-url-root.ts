import { baseURL } from './constant.js';
import { normalizeUrl } from './normalize-url.js';

/**
 * @example
 * https://example.org/abc?name=123&name=234' => 'https://example.org/abc'
 * @param url the original url
 */
export const getURLRoot = (url: string): string => {
  if (!url) {
    return '';
  }
  const myUri = new URL(url, baseURL);
  myUri.search = '';
  myUri.hash = '';
  return normalizeUrl(myUri.href, url);
};
