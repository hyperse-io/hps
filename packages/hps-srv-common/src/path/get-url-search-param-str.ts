import { baseURL } from './constant.js';

export const getURLSearchParamStr = (url: string): string => {
  if (!url) {
    return '';
  }
  const myUri = new URL(url, baseURL);
  return myUri.searchParams.toString();
};
