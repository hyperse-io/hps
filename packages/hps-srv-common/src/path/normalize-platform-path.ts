import { posix, sep } from 'node:path';

/**
 * This function normalizes the platform path.
 * On Windows, it replaces backslashes with forward slashes and removes the drive letter from the path.
 * On other platforms, it does not modify the path.
 *
 * @param pathStr - The path string to be normalized.
 * @param newSep - The new separator. default is `sep` for unit test.
 * @returns The normalized path string.
 */
export const normalizePlatformPath = (
  pathStr: string,
  newSep: string = sep
): string => {
  return pathStr.split(newSep).join(posix.sep);
};
