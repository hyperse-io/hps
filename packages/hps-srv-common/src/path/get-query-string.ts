import { getURLSearchParamStr } from './get-url-search-param-str.js';

export const getQueryString = (url: string, name: string): string | null => {
  const params = new URLSearchParams(getURLSearchParamStr(url));
  return params.get(name);
};
