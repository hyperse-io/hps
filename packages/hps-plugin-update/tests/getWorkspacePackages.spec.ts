import { join } from 'node:path';
import { getWorkspacePackages } from '../src/utils/getWorkspacePackages.js';
import { getDirname } from './test-utils.js';

describe('getWorkspacePackages()', () => {
  const fixtureCwd = getDirname(import.meta.url, 'fixtures');
  it('Should correct resolve workspace packages', async () => {
    const packages = await getWorkspacePackages(join(fixtureCwd, 'mono'));
    for (const [dir, meta] of packages) {
      expect(dir).toBeDefined();
      expect({
        meta,
      }).toMatchSnapshot();
    }
  });

  it('Should correct normalize single normal project', async () => {
    const packages = await getWorkspacePackages(
      join(fixtureCwd, 'mono/website')
    );
    for (const [dir, meta] of packages) {
      expect(dir).toBeDefined();
      expect({
        meta,
      }).toMatchSnapshot();
    }
  });

  it('Should correct resolve workspace packages for single repo', async () => {
    const packages = await getWorkspacePackages(join(fixtureCwd, 'single'));
    for (const [dir, meta] of packages) {
      expect(dir).toBeDefined();
      expect({
        meta,
      }).toMatchSnapshot();
    }
  });
});
