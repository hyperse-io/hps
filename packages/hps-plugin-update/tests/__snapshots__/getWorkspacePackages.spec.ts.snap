// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`getWorkspacePackages() > Should correct normalize single normal project 1`] = `
{
  "meta": {
    "dir": "/Users/tianyingchun/Documents/hyperse-io/hps-mono/examples/example-modular",
    "name": "@example/example-modular",
    "relativeDir": "examples/example-modular",
    "version": "0.0.2-next.3",
  },
}
`;

exports[`getWorkspacePackages() > Should correct normalize single normal project 2`] = `
{
  "meta": {
    "dir": "/Users/tianyingchun/Documents/hyperse-io/hps-mono/examples/example-single",
    "name": "@example/example-single",
    "relativeDir": "examples/example-single",
    "version": "0.0.2-next.3",
  },
}
`;

exports[`getWorkspacePackages() > Should correct normalize single normal project 3`] = `
{
  "meta": {
    "dir": "/Users/tianyingchun/Documents/hyperse-io/hps-mono/packages/hps",
    "name": "@hyperse/hps",
    "relativeDir": "packages/hps",
    "version": "0.1.0-next.3",
  },
}
`;

exports[`getWorkspacePackages() > Should correct normalize single normal project 4`] = `
{
  "meta": {
    "dir": "/Users/tianyingchun/Documents/hyperse-io/hps-mono/packages/hps-plugin-build",
    "name": "@hyperse/hps-plugin-build",
    "relativeDir": "packages/hps-plugin-build",
    "version": "0.0.1",
  },
}
`;

exports[`getWorkspacePackages() > Should correct normalize single normal project 5`] = `
{
  "meta": {
    "dir": "/Users/tianyingchun/Documents/hyperse-io/hps-mono/packages/hps-plugin-deploy",
    "name": "@hyperse/hps-plugin-deploy",
    "relativeDir": "packages/hps-plugin-deploy",
    "version": "0.0.1",
  },
}
`;

exports[`getWorkspacePackages() > Should correct normalize single normal project 6`] = `
{
  "meta": {
    "dir": "/Users/tianyingchun/Documents/hyperse-io/hps-mono/packages/hps-plugin-info",
    "name": "@hyperse/hps-plugin-info",
    "relativeDir": "packages/hps-plugin-info",
    "version": "0.1.0-next.3",
  },
}
`;

exports[`getWorkspacePackages() > Should correct normalize single normal project 7`] = `
{
  "meta": {
    "dir": "/Users/tianyingchun/Documents/hyperse-io/hps-mono/packages/hps-plugin-pack",
    "name": "@hyperse/hps-plugin-pack",
    "relativeDir": "packages/hps-plugin-pack",
    "version": "0.0.1",
  },
}
`;

exports[`getWorkspacePackages() > Should correct normalize single normal project 8`] = `
{
  "meta": {
    "dir": "/Users/tianyingchun/Documents/hyperse-io/hps-mono/packages/hps-plugin-serve",
    "name": "@hyperse/hps-plugin-serve",
    "relativeDir": "packages/hps-plugin-serve",
    "version": "0.0.1",
  },
}
`;

exports[`getWorkspacePackages() > Should correct normalize single normal project 9`] = `
{
  "meta": {
    "dir": "/Users/tianyingchun/Documents/hyperse-io/hps-mono/packages/hps-plugin-update",
    "name": "@hyperse/hps-plugin-update",
    "relativeDir": "packages/hps-plugin-update",
    "version": "0.0.1",
  },
}
`;

exports[`getWorkspacePackages() > Should correct normalize single normal project 10`] = `
{
  "meta": {
    "dir": "/Users/tianyingchun/Documents/hyperse-io/hps-mono/packages/hps-srv-mock",
    "name": "@hyperse/hps-srv-mock",
    "relativeDir": "packages/hps-srv-mock",
    "version": "0.0.1",
  },
}
`;

exports[`getWorkspacePackages() > Should correct normalize single normal project 11`] = `
{
  "meta": {
    "dir": "/Users/tianyingchun/Documents/hyperse-io/hps-mono/packages/hps-srv-rolldown",
    "name": "@hyperse/hps-srv-rolldown",
    "relativeDir": "packages/hps-srv-rolldown",
    "version": "0.0.1",
  },
}
`;

exports[`getWorkspacePackages() > Should correct normalize single normal project 12`] = `
{
  "meta": {
    "dir": "/Users/tianyingchun/Documents/hyperse-io/hps-mono/packages/hps-srv-rspack",
    "name": "@hyperse/hps-srv-rspack",
    "relativeDir": "packages/hps-srv-rspack",
    "version": "0.0.1",
  },
}
`;

exports[`getWorkspacePackages() > Should correct normalize single normal project 13`] = `
{
  "meta": {
    "dir": "/Users/tianyingchun/Documents/hyperse-io/hps-mono/crates/hps-srv-rspack-plugin-import",
    "name": "@hyperse/hps-srv-rspack-plugin-import",
    "relativeDir": "crates/hps-srv-rspack-plugin-import",
    "version": "0.0.1",
  },
}
`;

exports[`getWorkspacePackages() > Should correct normalize single normal project 14`] = `
{
  "meta": {
    "dir": "/Users/tianyingchun/Documents/hyperse-io/hps-mono/website",
    "name": "@hyperse/hps-website",
    "relativeDir": "website",
    "version": "0.0.1",
  },
}
`;

exports[`getWorkspacePackages() > Should correct resolve workspace packages 1`] = `
{
  "meta": {
    "dir": "/Users/tianyingchun/Documents/hyperse-io/hps-mono/examples/example-modular",
    "name": "@example/example-modular",
    "relativeDir": "examples/example-modular",
    "version": "0.0.2-next.3",
  },
}
`;

exports[`getWorkspacePackages() > Should correct resolve workspace packages 2`] = `
{
  "meta": {
    "dir": "/Users/tianyingchun/Documents/hyperse-io/hps-mono/examples/example-single",
    "name": "@example/example-single",
    "relativeDir": "examples/example-single",
    "version": "0.0.2-next.3",
  },
}
`;

exports[`getWorkspacePackages() > Should correct resolve workspace packages 3`] = `
{
  "meta": {
    "dir": "/Users/tianyingchun/Documents/hyperse-io/hps-mono/packages/hps",
    "name": "@hyperse/hps",
    "relativeDir": "packages/hps",
    "version": "0.1.0-next.3",
  },
}
`;

exports[`getWorkspacePackages() > Should correct resolve workspace packages 4`] = `
{
  "meta": {
    "dir": "/Users/tianyingchun/Documents/hyperse-io/hps-mono/packages/hps-plugin-build",
    "name": "@hyperse/hps-plugin-build",
    "relativeDir": "packages/hps-plugin-build",
    "version": "0.0.1",
  },
}
`;

exports[`getWorkspacePackages() > Should correct resolve workspace packages 5`] = `
{
  "meta": {
    "dir": "/Users/tianyingchun/Documents/hyperse-io/hps-mono/packages/hps-plugin-deploy",
    "name": "@hyperse/hps-plugin-deploy",
    "relativeDir": "packages/hps-plugin-deploy",
    "version": "0.0.1",
  },
}
`;

exports[`getWorkspacePackages() > Should correct resolve workspace packages 6`] = `
{
  "meta": {
    "dir": "/Users/tianyingchun/Documents/hyperse-io/hps-mono/packages/hps-plugin-info",
    "name": "@hyperse/hps-plugin-info",
    "relativeDir": "packages/hps-plugin-info",
    "version": "0.1.0-next.3",
  },
}
`;

exports[`getWorkspacePackages() > Should correct resolve workspace packages 7`] = `
{
  "meta": {
    "dir": "/Users/tianyingchun/Documents/hyperse-io/hps-mono/packages/hps-plugin-pack",
    "name": "@hyperse/hps-plugin-pack",
    "relativeDir": "packages/hps-plugin-pack",
    "version": "0.0.1",
  },
}
`;

exports[`getWorkspacePackages() > Should correct resolve workspace packages 8`] = `
{
  "meta": {
    "dir": "/Users/tianyingchun/Documents/hyperse-io/hps-mono/packages/hps-plugin-serve",
    "name": "@hyperse/hps-plugin-serve",
    "relativeDir": "packages/hps-plugin-serve",
    "version": "0.0.1",
  },
}
`;

exports[`getWorkspacePackages() > Should correct resolve workspace packages 9`] = `
{
  "meta": {
    "dir": "/Users/tianyingchun/Documents/hyperse-io/hps-mono/packages/hps-plugin-update",
    "name": "@hyperse/hps-plugin-update",
    "relativeDir": "packages/hps-plugin-update",
    "version": "0.0.1",
  },
}
`;

exports[`getWorkspacePackages() > Should correct resolve workspace packages 10`] = `
{
  "meta": {
    "dir": "/Users/tianyingchun/Documents/hyperse-io/hps-mono/packages/hps-srv-mock",
    "name": "@hyperse/hps-srv-mock",
    "relativeDir": "packages/hps-srv-mock",
    "version": "0.0.1",
  },
}
`;

exports[`getWorkspacePackages() > Should correct resolve workspace packages 11`] = `
{
  "meta": {
    "dir": "/Users/tianyingchun/Documents/hyperse-io/hps-mono/packages/hps-srv-rolldown",
    "name": "@hyperse/hps-srv-rolldown",
    "relativeDir": "packages/hps-srv-rolldown",
    "version": "0.0.1",
  },
}
`;

exports[`getWorkspacePackages() > Should correct resolve workspace packages 12`] = `
{
  "meta": {
    "dir": "/Users/tianyingchun/Documents/hyperse-io/hps-mono/packages/hps-srv-rspack",
    "name": "@hyperse/hps-srv-rspack",
    "relativeDir": "packages/hps-srv-rspack",
    "version": "0.0.1",
  },
}
`;

exports[`getWorkspacePackages() > Should correct resolve workspace packages 13`] = `
{
  "meta": {
    "dir": "/Users/tianyingchun/Documents/hyperse-io/hps-mono/crates/hps-srv-rspack-plugin-import",
    "name": "@hyperse/hps-srv-rspack-plugin-import",
    "relativeDir": "crates/hps-srv-rspack-plugin-import",
    "version": "0.0.1",
  },
}
`;

exports[`getWorkspacePackages() > Should correct resolve workspace packages 14`] = `
{
  "meta": {
    "dir": "/Users/tianyingchun/Documents/hyperse-io/hps-mono/website",
    "name": "@hyperse/hps-website",
    "relativeDir": "website",
    "version": "0.0.1",
  },
}
`;

exports[`getWorkspacePackages() > Should correct resolve workspace packages for single repo 1`] = `
{
  "meta": {
    "dir": "/Users/tianyingchun/Documents/hyperse-io/hps-mono/examples/example-modular",
    "name": "@example/example-modular",
    "relativeDir": "examples/example-modular",
    "version": "0.0.2-next.3",
  },
}
`;

exports[`getWorkspacePackages() > Should correct resolve workspace packages for single repo 2`] = `
{
  "meta": {
    "dir": "/Users/tianyingchun/Documents/hyperse-io/hps-mono/examples/example-single",
    "name": "@example/example-single",
    "relativeDir": "examples/example-single",
    "version": "0.0.2-next.3",
  },
}
`;

exports[`getWorkspacePackages() > Should correct resolve workspace packages for single repo 3`] = `
{
  "meta": {
    "dir": "/Users/tianyingchun/Documents/hyperse-io/hps-mono/packages/hps",
    "name": "@hyperse/hps",
    "relativeDir": "packages/hps",
    "version": "0.1.0-next.3",
  },
}
`;

exports[`getWorkspacePackages() > Should correct resolve workspace packages for single repo 4`] = `
{
  "meta": {
    "dir": "/Users/tianyingchun/Documents/hyperse-io/hps-mono/packages/hps-plugin-build",
    "name": "@hyperse/hps-plugin-build",
    "relativeDir": "packages/hps-plugin-build",
    "version": "0.0.1",
  },
}
`;

exports[`getWorkspacePackages() > Should correct resolve workspace packages for single repo 5`] = `
{
  "meta": {
    "dir": "/Users/tianyingchun/Documents/hyperse-io/hps-mono/packages/hps-plugin-deploy",
    "name": "@hyperse/hps-plugin-deploy",
    "relativeDir": "packages/hps-plugin-deploy",
    "version": "0.0.1",
  },
}
`;

exports[`getWorkspacePackages() > Should correct resolve workspace packages for single repo 6`] = `
{
  "meta": {
    "dir": "/Users/tianyingchun/Documents/hyperse-io/hps-mono/packages/hps-plugin-info",
    "name": "@hyperse/hps-plugin-info",
    "relativeDir": "packages/hps-plugin-info",
    "version": "0.1.0-next.3",
  },
}
`;

exports[`getWorkspacePackages() > Should correct resolve workspace packages for single repo 7`] = `
{
  "meta": {
    "dir": "/Users/tianyingchun/Documents/hyperse-io/hps-mono/packages/hps-plugin-pack",
    "name": "@hyperse/hps-plugin-pack",
    "relativeDir": "packages/hps-plugin-pack",
    "version": "0.0.1",
  },
}
`;

exports[`getWorkspacePackages() > Should correct resolve workspace packages for single repo 8`] = `
{
  "meta": {
    "dir": "/Users/tianyingchun/Documents/hyperse-io/hps-mono/packages/hps-plugin-serve",
    "name": "@hyperse/hps-plugin-serve",
    "relativeDir": "packages/hps-plugin-serve",
    "version": "0.0.1",
  },
}
`;

exports[`getWorkspacePackages() > Should correct resolve workspace packages for single repo 9`] = `
{
  "meta": {
    "dir": "/Users/tianyingchun/Documents/hyperse-io/hps-mono/packages/hps-plugin-update",
    "name": "@hyperse/hps-plugin-update",
    "relativeDir": "packages/hps-plugin-update",
    "version": "0.0.1",
  },
}
`;

exports[`getWorkspacePackages() > Should correct resolve workspace packages for single repo 10`] = `
{
  "meta": {
    "dir": "/Users/tianyingchun/Documents/hyperse-io/hps-mono/packages/hps-srv-mock",
    "name": "@hyperse/hps-srv-mock",
    "relativeDir": "packages/hps-srv-mock",
    "version": "0.0.1",
  },
}
`;

exports[`getWorkspacePackages() > Should correct resolve workspace packages for single repo 11`] = `
{
  "meta": {
    "dir": "/Users/tianyingchun/Documents/hyperse-io/hps-mono/packages/hps-srv-rolldown",
    "name": "@hyperse/hps-srv-rolldown",
    "relativeDir": "packages/hps-srv-rolldown",
    "version": "0.0.1",
  },
}
`;

exports[`getWorkspacePackages() > Should correct resolve workspace packages for single repo 12`] = `
{
  "meta": {
    "dir": "/Users/tianyingchun/Documents/hyperse-io/hps-mono/packages/hps-srv-rspack",
    "name": "@hyperse/hps-srv-rspack",
    "relativeDir": "packages/hps-srv-rspack",
    "version": "0.0.1",
  },
}
`;

exports[`getWorkspacePackages() > Should correct resolve workspace packages for single repo 13`] = `
{
  "meta": {
    "dir": "/Users/tianyingchun/Documents/hyperse-io/hps-mono/crates/hps-srv-rspack-plugin-import",
    "name": "@hyperse/hps-srv-rspack-plugin-import",
    "relativeDir": "crates/hps-srv-rspack-plugin-import",
    "version": "0.0.1",
  },
}
`;

exports[`getWorkspacePackages() > Should correct resolve workspace packages for single repo 14`] = `
{
  "meta": {
    "dir": "/Users/tianyingchun/Documents/hyperse-io/hps-mono/website",
    "name": "@hyperse/hps-website",
    "relativeDir": "website",
    "version": "0.0.1",
  },
}
`;
